azavea/echo-locator
# Feature/kak/toggle ecc#129

## Overview

Add a toggle button to the map sidebar form to show/hide non-Expanded Choice Community zip codes.


### Demo

Checked:
![image](https://user-images.githubusercontent.com/960264/56511735-56f55b80-64fb-11e9-811d-d0cc2f0fd230.png)

Unchecked:
![image](https://user-images.githubusercontent.com/960264/56511763-62e11d80-64fb-11e9-96e5-ab3523e74c83.png)

Route, details, and start and end markers continue to display in detail view if box is unchecked while viewing a non-ECC zipcode:

![image](https://user-images.githubusercontent.com/960264/56511828-8efc9e80-64fb-11e9-90f5-e380f1321860.png)


### Notes

Moving the neighborhood bounds layer to a layer group of custom vector grids was necessary to make the layer dynamic.


## Testing Instructions

 * Use non-ECC checkbox should default to checked on page reload or navigation to map
 * Unchecking box should hide non-ECC bounds and centroid markers on map
 * Unchecking box should exclude non-ECC neighborhood results from sidebar
 * Unchecking box while in details for non-ECC zipcode should not hide route or details
 * Marker highlighting for sidebar neighborhoods should continue to work as expected


Resolves #129
